src/formats/TRR.cpp

```diff
@@ -82,33 +87,88 @@ TRRFormat::TRRFormat(std::string path, File::Mode mode, File::Compression compre
     } else if (mode == File::APPEND) {
         try {
             determine_frame_offsets();
-        } catch (const Error&) {  // NOLINT(bugprone-empty-catch)
+        } catch (const Error&) { // NOLINT(bugprone-empty-catch)
             // Ignore exceptions, because the file might not exist. If it does,
             // we need to get the number of atoms and frames for appending.
         }
     }
 }
 
-size_t TRRFormat::size() {
-    return frame_positions_.size();
-}
+size_t TRRFormat::size() { return frame_positions_.size(); }
 
 void TRRFormat::read_at(size_t index, Frame& frame) {
     index_ = index;
     file_.seek(frame_positions_[index_]);
     read(frame);
 }
 
+template <typename T>
+typename std::enable_if<std::is_same<T, float>::value || std::is_same<T, double>::value, void>::type
+read_real_vec(XDRFile& file, std::vector<T>& dx) {
+    if constexpr (sizeof(T) == 4) {
+        file.read_f32(dx);
+    } else {
+        static_assert(sizeof(T) == 8);
+        file.read_f64(dx);
+    }
+}
```

Comment:
I think I prefer template specialization for this kind of things:

```suggestion
template <typename T>
void read_real_vec(XDRFile& file, std::vector<T>& dx) {
    throw std::runtime_error("read_real_vec can only be used with float or double")
}

template <>
void read_real_vec<double>(XDRFile& file, std::vector<double>& dx) {
    file.read_f64(dx);
}

template <>
void read_real_vec<float>(XDRFile& file, std::vector<float>& dx) {
    file.read_f32(dx);
}
```

Reply:
Thanks for the suggestion.
I've made it a compile-time error instead of a runtime error by using `static_assert` instead of exceptions.

Reply:
good idea!